Add device monitoring (#6)

kev1pic opened the pull request:
Hi there,

I am using gtkterm for embedded development, and I was only missing autoreconnect feature (saw it in issue #6 too), so here is a suggested solution.

Known limitation: 
Using thread involves concurrent access to `struct configuration_port config`. So in theory, as it is global variable it is pretty hard work to add mutex. In practise probabilty is relatively low.

It would be great to get your feedback on it ;-)

wvdakker commented:
Hi,
Thanks for the patch. Looks good.
One (actually two) questions:
- Can you add a copyright header in your source?
  This makes it easier for me to include it into Debian
- Can you make a pull request against the autoreconnect branch?
  I have created it for testing this code. 

Many thanks!

kev1pic commented:
Thanks for review.

I added copyright and changed pull request on autoreconnect branch.
Regarding copyright, not sure if there are any conventions, so I copy-paste from other files and I added my name. Let me know if it's convenient.

wvdakker commented:
I have merged it with the autoreconnect branch. I also will test it.
I think the mutex must be solved before we can merge it with the master branch.
I also will look at that later this month.

Thanks for sharing!

wvdakker commented:
Hi Kevin,
I have tested the patch.
First time it did not work and I was wondering where the 'device_monitor_start' is called.
First I thought it was a callback from the udev lib, but couldnt find any hooks.
After that I called 'device_monitor_start' from gtkmain.c and it worked.

Did you forget to include the gtkmain.c patch? Or did I miss something?

kev1pic commented:
Hi,
You are right, `device_monitor_start` is missing. I originally included it just before `gtk_main()` in gktterm.c file.
I probably made a mistake while commiting, sorry...

wvdakker commented:
Added.
